x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/topn/TopNOperator.java

```diff
@@ -405,7 +399,17 @@ private Iterator<Page> toPages() {
 
                 p++;
                 if (p == size) {
-                    result.add(new Page(Arrays.stream(builders).map(ResultBuilder::build).toArray(Block[]::new)));
+                    Block[] blocks = new Block[builders.length];
```

Comment:
I thought we should use `Block.Builder.buildAll`, but these builders are ResultBuilder.

Reply:
Yeah! But those aren't actually builders.....